TestFibermap test_gather_targetphot unit test failure on Perlmutter

As logged in https://data.desi.lbl.gov/desi/spectro/redux/dailytest/log/perlmutter/desispec_20230602.log, desispec unit test `TestFibermap.test_gather_targetphot` failed on Perlmutter last night:
```
=================================== FAILURES ===================================
_____________________ TestFibermap.test_gather_targetphot ______________________

self = <desispec.test.test_photo.TestFibermap testMethod=test_gather_targetphot>

    @unittest.skipUnless(standard_nersc_environment, "not at NERSC")
    def test_gather_targetphot(self):
        """Test that we get the correct Tractor photometry for an input set of objects."""
    
>       tractorphot = gather_tractorphot(self.input_cat)

py/desispec/test/test_photo.py:81: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
py/desispec/io/photo.py:947: in gather_tractorphot
    bugout = _gather_tractorphot_onebrick(input_cat[bug], dr9dir, radius_match, racolumn, deccolumn)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

input_cat = <Table length=3>
     TARGETID        TARGET_RA        TARGET_DEC    ... BRICK_OBJID PHOTSYS
      int64           flo...06 1.40073953879623 ...           0        
43977491483722156      150.1306899         1.505752 ...           0        
dr9dir = '/global/cfs/cdirs/desi/external/legacysurvey/dr9', radius_match = 1.0
racolumn = 'TARGET_RA', deccolumn = 'TARGET_DEC'

    def _gather_tractorphot_onebrick(input_cat, dr9dir, radius_match, racolumn, deccolumn):
        """Support routine for gather_tractorphot."""
    
>       assert(np.all(input_cat['BRICKNAME'] == input_cat['BRICKNAME'][0]))
E       AssertionError

py/desispec/io/photo.py:739: AssertionError
```
This could be related to #2057. @sbailey please triage this? @weaverba137 FYI.

--- a/py/desispec/io/photo.py
+++ b/py/desispec/io/photo.py
@@ -944,9 +944,10 @@ def gather_tractorphot(input_cat, racolumn='TARGET_RA', deccolumn='TARGET_DEC',
             input_cat['BRICK_OBJID'][bug] = 0
             input_cat['PHOTSYS'][bug] = ''
 
-            bugout = _gather_tractorphot_onebrick(input_cat[bug], dr9dir, radius_match, racolumn, deccolumn)
-            for col in bugout.colnames:
-                out[col][bug] = bugout[col]
+            bugout = Table(np.hstack(np.repeat(tractorphot_datamodel(), len(bug))))
+            for onebrickname in set(input_cat['BRICKNAME'][bug]):
+                I = np.where(onebrickname == input_cat['BRICKNAME'][bug])[0]
+                bugout[I] = _gather_tractorphot_onebrick(input_cat[bug][I], dr9dir, radius_match, racolumn, deccolumn)
             
     if columns is not None:
         if type(columns) is not list:

